spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/data/neo4j/Neo4jDataAutoConfigurationTests.java

```diff
@@ -139,6 +142,25 @@ public void eventListenersAreAutoRegistered() {
 				.onPreSave(any(Event.class));
 	}
 
+	@Test
+	public void autoIndexConfiguration() {
+		load(CustomConfigurationFactory.class);
+		assertThat(this.context.getBean(org.neo4j.ogm.config.Configuration.class)
+				.getAutoIndex()).isEqualTo(AutoIndexMode.NONE);
+
+		load(CustomConfigurationFactory.class, "spring.data.neo4j.auto-index=assert");
```

Comment:
I want to check that the `Neo4jProperties` instance will correctly create the Configuration bean `in createConfiguration` regarding the configuration properties values.